### PR TITLE
[Snap]: Added Display QR core feature

### PR DESCRIPTION
**Pull request by AnthonyLaw**
## What was the issue?
- Receive button feature is missing.

## What's the fix?
- Added generation hash seed in the network state, it is used to generate QR code content.
- Installed qrcode, symbol-sdk dependencies.
- Added Receive a modal box
- Implement the Receive modal box in the Receive button.

## Screenshot:

Screenshot 2024-06-26 at 11 58 00 PM

### CODE REVIEW

File: snap/frontend/package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
   "dependencies": {
     "@metamask/detect-provider": "^2.0.0",
     "next": "14.1.3",
+    "qrcode": "^1.5.3",
```
Comment (Wayonb) [inline snap/frontend/package.json:25]: Did we standardize on a  QR code?

**Review by Wayonb — APPROVED** [review]
👍🏾

File: snap/frontend/package.json (hunk 1 above)
Comment (AnthonyLaw) [reply snap/frontend/package.json:25]: I don't think we have **official** standardize QR code, but we did have the library [symbol-qr](https://github.com/symbol/qr-library) (old library, used by our mobile / desktop wallet app)

**Comment by AnthonyLaw** [comment]
@Wayonb rebased.

**Review by Wayonb — APPROVED** [review]
👍🏾

**Pull request merged**